### PR TITLE
[15.0][ADD] New modules for Ecuadorian Localization

### PR DESCRIPTION
**Pull request by celm1990**
Hi, the present PR is contribution of various people's

Roadmap is here https://jhonromero26.notion.site/Localizaci-n-Ecuatoriana-a-Odoo-15-d3f3fcec7f404c2ead41bbf4b1a9539f

Currently PR has module for create, sign and send xml to SRI for Customer invoice, Customer Credit Note, Customer Debit Note, Purchase Liquidation. Coming soon Withholding, Delivery Notes and Import Documents from Txt downloaded from SRI

Any feedback is welcome

### CODE REVIEW

**Review by dreispt — APPROVED** [review]
Please remove the top level __init__.py file.
Other than that, looks good.

**Comment by celm1990** [comment]
> Please remove the top level **init**.py file. Other than that, looks good.

@dreispt thanks for your review, I updated code, any additional comments let us know

File: l10n_ec_account_edi/models/account_edi_format.py [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+                _(
+                    "You must set Emission address into Journal: %s",
+                    journal.display_name,
+                )
+            )
+        # direccion matriz
+        if not contact_address.commercial_partner_id.street:
+            errors.append(
+                _(
+                    "You must set Street into Company: %s",
```
Comment (cormaza) [inline l10n_ec_account_edi/models/account_edi_format.py]: This message should be clear, is emission address without street assigned

File: l10n_ec_account_edi/models/account_edi_format.py [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+                    contact_address.commercial_partner_id.display_name,
+                )
+            )
+        # direccion de establecimiento
+        if not contact_address.street:
+            errors.append(
+                _(
+                    "You must set Street into Contact: %s",
```
Comment (cormaza) [inline l10n_ec_account_edi/models/account_edi_format.py]: Same above

File: l10n_ec_account_edi/models/account_edi_format.py [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+                            _(
+                                "You must set Tax Code for Electronic Documents into Tax: %s",
+                                tax.display_name,
+                            )
+                        )
+            # forma de pago por defecto
+            if not journal.l10n_ec_sri_payment_id:
+                errors.append(
+                    _(
+                        "You must set Payment Method SRI on Journal: %s",
```
Comment (cormaza) [inline l10n_ec_account_edi/models/account_edi_format.py]: This should be validated only if move doesn't have payment method asigned, doesn't make sense restrict if move have assigned one

**Review by cormaza — CHANGES_REQUESTED** [review]
Functional test:

- On demo data should be added payment method by default
image

- Credit note creation should add as required reason field on wizard, because when credit note is created without this value you cannot modify, I tested with fully refund case
image

**Comment by celm1990** [comment]
Hi @cormaza 
Thanks you for your review, my coments inline
> * On demo data should be added payment method by default

Sorry but this change is complicated because field is created on [odoo base](https://github.com/odoo/odoo/blob/453deaec268425a4a8a8c03ebae82e024b564507/addons/l10n_ec/models/account_move.py#L132-L135) so we try add  default value on this field for new records, please see [this line](https://github.com/Odoo-EC/l10n-ecuador/blob/bfadcd1e181985e1531cf8769641ebc5c1ec5303/l10n_ec_account_edi/models/account_move.py#L15)
however on data demo is not take into account for current records.

A alternative is create `post_install` script but personally consider irrelevant  for demo purpose, feel free to propose your PR to improve this here or Odoo base(if you have other alternative i like meet)

Other changes are done, please can you update your review

**Comment by cormaza** [comment]
> Hi @cormaza Thanks you for your review, my coments inline
> 
> > * On demo data should be added payment method by default
> 
> Sorry but this change is complicated because field is created on [odoo base](https://github.com/odoo/odoo/blob/453deaec268425a4a8a8c03ebae82e024b564507/addons/l10n_ec/models/account_move.py#L132-L135) so we try add default value on this field for new records, please see [this line](https://github.com/Odoo-EC/l10n-ecuador/blob/bfadcd1e181985e1531cf8769641ebc5c1ec5303/l10n_ec_account_edi/models/account_move.py#L15) however on data demo is not take into account for current records.
> 
> A alternative is create `post_install` script but personally consider irrelevant for demo purpose, feel free to propose your PR to improve this here or Odoo base(if you have other alternative i like meet)
> 
> Other changes are done, please can you update your review

IMO if you will change a field to required on any view, if you don't make a script to handle old data can be messe up system for old data, because this is a change for this localization, should be good to have this script on this release

**Comment by celm1990** [comment]
> IMO if you will change a field to required on any view, if you don't make a script to handle old data can be messe up system for old data, because this is a change for this localization, should be good to have this script on this release

@cormaza I added script to set default payment on current records, please update your review

**Comment by cormaza** [comment]
Messages should be more clear, example

image

There are two message with same mean, should be more clear, for example if for customer, or for journal address, etc

**Comment by celm1990** [comment]
> There are two message with same mean, should be more clear, for example if for customer, or for journal address, etc

@cormaza done, thanks for observation
`dirEstablecimiento` is required and taken from journal's Emission address( as Branch address) see [this line](https://github.com/Odoo-EC/l10n-ecuador/blob/9ca8b864b99fa6d61b0af8a4c262207116e2e17f/l10n_ec_account_edi/models/account_edi_document.py#L376)
`direccionComprador` is optional, if empty so NA is by default see [this line](https://github.com/Odoo-EC/l10n-ecuador/blob/9ca8b864b99fa6d61b0af8a4c262207116e2e17f/l10n_ec_account_edi/models/account_edi_document.py#L388)

So validation for field `contact_address.commercial_partner_id.street` is innecesary and remove

**Review by cormaza — APPROVED** [review]
Functional and code review ok!!!
LGTM

**Comment by jorge-luis-dev** [comment]
I review and approved

**Review by jorge-luis-dev — APPROVED** [review]

**Review by jvinueza — APPROVED** [review]

**Review by agsistemas — APPROVED** [review]
I review and approved

**Review by yazber — APPROVED** [review]

**Comment by cormaza** [comment]
@pedrobaeza can you help us to merge this one? to continue on roadmap?

**Comment by celm1990** [comment]
Thanks to all for reviews and comments, new features are coming soon, all contributions are welcome 
@dreispt  you can set 15.0 as default branch please?

**Pull request merged**